modules/fedimint-mint-tests/tests/tests.rs

```diff
@@ -235,6 +249,53 @@ mod fedimint_migration_tests {
         dbtx.insert_new_entry(&backup_key, &ecash_backup).await;
     }
 
+    async fn create_client_db_with_v0_data(mut dbtx: DatabaseTransaction<'_>) {
+        dbtx.insert_new_entry(&DatabaseVersionKey, &DatabaseVersion(0))
+            .await;
+
+        let (_, pubkey) = secp256k1::generate_keypair(&mut OsRng);
+        let keypair = KeyPair::new_global(&mut OsRng);
+
+        let msg = Message::from_bytes(b"Hello World!");
+        let threshold = 3;
+
+        let bkey = BlindingKey::random();
+        let bmsg = blind_message(msg, bkey);
+
+        let (_, _pks, sks) = dealer_keygen(threshold, 5);
+
+        let bsig_shares = sks
+            .iter()
+            .map(|sk| sign_blinded_msg(bmsg, *sk))
+            .collect::<Vec<BlindedSignatureShare>>();
+
+        let bsig_shares = (1_u64..)
+            .zip(bsig_shares)
+            .take(5)
+            .collect::<BTreeMap<u64, BlindedSignatureShare>>();
+
+        let bsig = aggregate_signature_shares(&bsig_shares);
+        let sig = unblind_signature(bkey, bsig);
+
+        dbtx.insert_new_entry(
+            &NoteKey {
+                amount: Amount::from_sats(1000),
+                nonce: Nonce(pubkey),
+            },
+            &SpendableNote {
+                signature: sig,
+                spend_key: keypair,
+            },
+        )
+        .await;
+
+        dbtx.insert_new_entry(&NextECashNoteIndexKey(Amount::from_sats(1000)), &3)
+            .await;
+
+        dbtx.insert_new_entry(&CancelledOOBSpendKey(OperationId(BYTE_32)), &())
+            .await;
```

Comment:
Restore state is likely being added later?

Reply:
Yes, added in commit after rebase